rate_limit/rate_limit.py

```diff
@@ -58,6 +58,19 @@ def __init__(self, app, **conf):
         # Backend is used to store count of requests.
         self.backend_host = self.__conf.get('backend_host', '127.0.0.1')
         self.backend_port = common.to_int(self.__conf.get('backend_port'), 6379)
+
+        # Load password from configuration file
+        self.backend_password = None
+        self.backend_secret_file = self.__conf.get('backend_secret_file')
+        if self.backend_secret_file:
+            try:
+                with open(self.backend_secret_file, 'r') as f:
+                    self.backend_password = f.read().strip()
+            except IOError as e:
+                self.logger.error(
+                    "error loading backend secret from file '%s': %s" % (self.backend_secret_file, e)
```

Comment:
Please let the logging class do the string formatting.

Reply:
...okay, you were right in our call. It's a custom logging class that doesn't support kwargs. I'd put that on the refactor list but then this is fine the way it is. I'd probably go with an f-string but whatever. :)